netbench/netbench-test-player/src/endpoint.rs

```diff
@@ -0,0 +1,62 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+use crate::status::{Status, StatusTracker};
+use netbench::collector::{run, Args, RunHandle};
+
+use strum_macros::EnumString;
+
+/// For the purposes of coordination, what is the kind of endpoint we are?
+#[derive(Debug, PartialEq, Clone, Copy, EnumString, Hash)]
+#[strum(ascii_case_insensitive)]
```

Comment:
What is strum and why do we need it?

Reply:
Note the `derive(...EnumString, ...)`. This outputs a TryFrom<str> implementation for this struct. The `strum(ascii_case_insensitive)` is a option affecting the implementation of that gets generated for parsing.